### PR TITLE
Fix listCollections dispatch on Add Rules page

### PR DESCRIPTION
**Pull request by dopeters**
1. On add rules page, we are not checking for errors when determining if an action has been dispatches which can lead to endless rerender. Added that logic and make object destructing more clear.

2. We need a way to retrieve all collections on the add rules page for populating the dropdown. (if a time filter is applied, only collections with active granules will populate that dropdown). Added a `listAll` property to the options passed to `listCollections`

### CODE REVIEW

File: app/src/js/actions/index.js [hunk 1]
```diff
@@ -128,16 +128,17 @@ export const checkApiVersion = () => {
 };
 
 export const listCollections = (options) => {
+  const { listAll, ...queryOptions } = options;
   return (dispatch, getState) => {
     const timeFilters = fetchCurrentTimeFilters(getState().datepicker);
-    const urlPath = `collections${isEmpty(timeFilters) ? '' : '/active'}`;
+    const urlPath = `collections${isEmpty(timeFilters) || listAll ? '' : '/active'}`;
```
Comment (dopeters) [inline app/src/js/actions/index.js:134]: Can test that the endless rerender is no longer happening by changing `/active` here to something that doesn't exist

**Comment by flamingbear** [comment]
I know it must have been a different PR, but will you pin the @cumulus/api to exactly 1.21.1-alpha.2? instead of ^1.21.1-alpha.2.  We will thank ourselves later I'm sure.

File: app/src/js/actions/index.js (hunk 1 above)
Comment (flamingbear) [reply app/src/js/actions/index.js:134]: I was able to install 1.21.1-alpha.1 and verify this no longer occurs as well.

File: app/src/js/components/Rules/add.js [hunk 2]
```diff
@@ -108,11 +108,14 @@ const AddRule = ({
     }
   }, [name, rulesMap, isCopy]);
 
-  const dispatched = ({ list: { inflight, meta: { queriedAt } } }) =>
-    inflight || queriedAt !== undefined;
+  const dispatched = ({ list }) => {
+    const { inflight, meta, error } = list || {};
+    const { queriedAt } = meta || {};
+    return error || inflight || queriedAt !== undefined;
+  };
```
Comment (flamingbear) [inline app/src/js/components/Rules/add.js:115]: 💯

**Pull request merged**